mcs/class/referencesource/System.Data.Entity/System/Data/Objects/ELinq/MethodCallTranslator.cs

```diff
@@ -1008,6 +1014,13 @@ internal IndexOfTranslator()
                 private static IEnumerable<MethodInfo> GetMethods()
                 {
                     yield return typeof(String).GetMethod("IndexOf", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(string) }, null);
+#if MONO
+                    yield return typeof(String).GetMethod("IndexOf", BindingFlags.Public | BindingFlags.Instance, null, new Type[] { typeof(char) }, null);
```

Comment:
Remove methods which take more than 1 argument, the code can handle only `1 argument for String.IndexOf`

Reply:
Done